CIPs/cip-granda_mento.md

```diff
@@ -0,0 +1,187 @@
+2. **`function approveExchangeProposal(uint256 id) external`** - Approves a proposed exchange.
+   * Only callable by the approver address.
+   * Marks the proposed exchange with the given `id` as approved.
+   * Records:
+     * The exchange as in the Approved state.
+     * The timestamp at which the approval occurred (ie `block.timestamp`).
+3. **`function cancelExchangeProposal(uint256 id) external`** - Cancels a proposed exchange.
+   * The permitted caller depends upon the state of the exchange proposal:
+     * If in the Proposed state, can only be called by the proposer.
+     * If in the Approved state, can only be called by Governance.
+     * If in any other state, cannot be called.
+   * Refunds the proposed exchange's deposited sell asset to the proposer.
+   * Records:
+     * The exchange as in the Cancelled state.
+4. **`function executeExchange(uint256 id) external`** - Executes an exchange.
+   * Callable by anyone.
+   * Requires the exchange proposal to have been approved.
+   * Requires the required waiting period for an approved exchange to have elapsed since the time it was approved (ie `block.timestamp - approval timestamp >= exchangeWaitPeriod`)
+   * Makes the exchange:
+     * If selling CELO and purchasing stable token:
+       * The CELO is sent to the Reserve.
+       * Stable token is minted to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+     * If selling stable token and purchasing CELO:
+       * The stable token is burned.
+       * CELO is transferred from the Reserve to the exchange proposer according to the rate originally recorded when the exchange was proposed.
+   * Records:
+     * The exchange as in the Executed state.
+5. **`function setStableTokenExchangeLimits(address stableToken, uint256 minExchangeAmount, uint256 maxExchangeAmount) external`**
+   * Only callable by Governance.
+   * Sets the minimum and maximum amounts of stable token that can be minted/burned in an exchange by updating `stableTokenExchangeLimits[stableToken]`.
+     * Effectively "enables" exchanging a stable token if the min & max amounts were previously 0.
+     * Effectively "disables" exchanging a stable token if setting the min & max amounts to 0.
+
+## Rationale
+
+Granda Mento is not intended to undermine the market or provide a way for large players to "dump" CELO for cXXX. It's also not intended to be an exceptionally elegant or complicated solution-- instead, this is looking to be something that is simple, straightforward, and works alongside the existing implementation of Mento. Any larger-scale Mento design modifications to facilitate large stable token mints are out of scope, and should be considered for a medium/long term design change to the stability protocol. The proposed implementation attempts to navigate the tradeoffs of complexity, centralization, economic safety, and ease of use for all actors in the Celo ecosystem.
```

Comment:
I might actually call out here that it should be intended to be much simpler or understandable than some of the comparable solutions for users. This would summarize of the comparative analysis and justify a few of the design decisions above.

vs OTC: Fees are fixed, constant, and lower
vs. Exchange Limit Orders: Fill or cancellation time is a predictible
vs. Trading bot: Fixed slippage, single transaction execution, less technical knowledge needed